pymongo/pool.py

```diff
@@ -808,7 +860,7 @@ def _hello(self, cluster_time, topology_version, heartbeat_frequency):
             self.generation = self.pool_gen.get(self.service_id)
         return hello
 
-    def _next_reply(self):
+    def _next_reply(self) -> Mapping[str, Any]:
```

Comment:
Let's change this to `Dict[str, Any]`, then we can remove the covariant=True.

Reply:
Sounds good, done.